crates/rpc/src/v02/types.rs

```diff
@@ -555,33 +700,41 @@ pub mod reply {
                     }
                 }
                 GatewayTransaction::Declare(GatewayDeclare::V0(txn)) => {
-                    Self::Declare(DeclareTransaction {
-                        common: CommonTransactionProperties {
+                    Self::Declare(DeclareTransaction::V1(DeclareTransactionV1 {
+                        common: CommonDeclareInvokeTransactionProperties {
                             hash: txn.transaction_hash,
                             max_fee: txn.max_fee,
-                            version: TransactionVersion::ZERO,
```

Comment:
So we're transforming the version from zero to one here? That does not seem to be correct.